### DISCUSSION TITLE
How to configure the basic network？

### DISCUSSION THREAD

**Discussion by Phoenix1017** [AltraMayor/gatekeeper#568] [General]
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: enp131s0f0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:0c:29:73:17:f6 brd ff:ff:ff:ff:ff:ff
    inet 192.168.111.185/24 brd 192.168.111.255 scope global enp131s0f0
       valid_lft forever preferred_lft forever
    inet6 fe80::20c:29ff:fe73:17f6/64 scope link
       valid_lft forever preferred_lft forever
3: ens192: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:0c:29:73:17:00 brd ff:ff:ff:ff:ff:ff
    inet 192.168.111.186/24 brd 192.168.111.255 scope global ens192
       valid_lft forever preferred_lft forever
    inet6 fe80::20c:29ff:fe73:1700/64 scope link
       valid_lft forever preferred_lft forever

lshw -c network -businfo
Bus info          Device      Class      Description
====================================================
pci@0000:03:00.0  enp131s0f0  network    VMXNET3 Ethernet Controller
pci@0000:0b:00.0  ens192      network    VMXNET3 Ethernet Controller

 cat lua/if_map.lua
return {
        ["enp131s0f0"] = "0000:03:00.0",
        ["ens192"] = "0000:0b:00.0",
}

cat debian/envvars
DPDK_ARGS=""
GATEKEEPER_ARGS="--lua-base-dir /etc/gatekeeper --log-base-dir /var/log/gatekeeper"
GATEKEEPER_INTERFACES="03:00.0 0b:00.0"

root@root:~/gatekeeper# ./build/gatekeeper -c 2 -n 4
EAL: Detected 4 lcore(s)
EAL: Detected 1 NUMA nodes
EAL: Multi-process socket /var/run/dpdk/rte/mp_socket
EAL: Selected IOVA mode 'PA'
EAL: Probing VFIO support...
EAL: VFIO support initialized
EAL:   Invalid NUMA socket, default to 0
EAL:   Invalid NUMA socket, default to 0
EAL: No legacy callbacks, legacy socket not created
EAL: See files in . for further log

cat gatekeeper_2022_06_16_03_01.log
Main/1: cycles/second = 2000002440, cycles/millisecond = 2000002, picosec/cycle = 499
Main/1: config: ./lua/gatekeeper/staticlib.lua:505: There is no map for enp133s0f0 in the front interface configuration
Main/1: Failed to configure Gatekeeper

**Comment by AltraMayor** [comment]
The log entry `Main/1: config: ./lua/gatekeeper/staticlib.lua:505: There is no map for enp133s0f0 in the front interface configuration` points out that you haven't edited the file `lua/net.lua` since the example interface `enp133s0f0` hasn't been changed to match your NICs.

Moreover, the example configuration requires 7 lcores to run on a single NUMA node (i.e. single physical processor), and 10 lcores to run on two NUMA nodes, but DPDK is only finding 4: `EAL: Detected 4 lcore(s)`.

You don't need to edit anything in the folder `debian/`. This folder is only used to create a Debian package for Gatekeeper.

**Comment by Phoenix1017** [comment]
您好，已收到亲的邮件。我将尽快查阅。-------------------------------------------------------市场调查合作伙伴！负责整个深圳区域的执行。黄楷杰Tel: 13662628489

**Comment by gmahajan1207** [comment]
hey ,
can anyone help me install the gatekeepr

**Comment by Phoenix1017** [comment]
您好，已收到亲的邮件。我将尽快查阅。-------------------------------------------------------市场调查合作伙伴！负责整个深圳区域的执行。黄楷杰Tel: 13662628489

**Comment by AltraMayor** [comment]
Hi @gmahajan1207,

The page [Tips for Deployments](https://github.com/AltraMayor/gatekeeper/wiki/Tips-for-Deployments) of our wiki has a full example you can adjust to your needs.
